app.py

```diff
@@ -336,11 +336,31 @@ def list_data_objects(**kwargs):
     # Build the query. If multiple criteria are specified, returned objects
     # should match all of the provided criteria (logical AND).
     query = {'query': {}}
-    if 'page_token' in req_body:
+    if 'page_token' in req_body:  # for paging
         query['from'] = req_body['page_token'] or 0
-    if 'alias' in req_body:
-        query['query']['term'] = {}
-        query['query']['term']['aliases.keyword'] = req_body['alias']
+    if 'alias' in req_body or 'checksum' in req_body or 'url' in req_body:
+        query['query']['bool'] = {'filter': []}
+        # Only fileMd5sum stored in Azul
+        if 'checksum_type' in req_body and req_body['checksum_type'].lower() != 'md5':
+            return {'data_objects': []}
```

Comment:
I don't know too much about what this code is even supposed to do, but why is this short circuiting and returning nothing if the checksum type isn't correct?

Reply:
Should there also be like an error message or something?

Reply:
I think this behavior is fine - Azul only stores MD5s, so if the provided checksum is not an MD5, then there will necessarily be no results. I don't think this needs an error because a request specifying a non-MD5 checksum type isn't necessarily wrong - the only error I could think to raise is a bad request error, but that doesn't seem correct to me since the request was well-formed.

Reply:
Sounds good, but what about if Azul were to support other kinds of checksums? Why not just let the query go through and return empty results?